app/controllers/cards_controller.rb

```diff
@@ -1,42 +1,39 @@
 class CardsController < ApplicationController
-  before_action :load_card, only: [:show, :edit, :update, :destroy, :crop_image]
+  before_action :load_card, only: [:edit, :update, :destroy, :crop_image]
+  before_action :load_deck
 
   def index
-    @card = current_user.cards.all
+    @cards = @deck.cards.all.page(params[:page]).per(4)
   end
 
   def new
-    @card = current_user.cards.new
+    @card = @deck.cards.new
   end
 
   def create
-    @card = current_user.cards.new(card_params)
-    if @card.save
-      flash[:notice] = 'Новая карта создана'
-      redirect_to new_card_path
-    else
-      flash[:notice] = 'Ошибка в создании карты'
-      render :new
+    respond_with(@card = @deck.cards.create(card_params.merge(user_id: current_user.id))) do |format|
+      format.html { redirect_to new_deck_card_path } if @card.save
     end
   end
 
   def update
     if @card.update(card_params)
-      redirect_to cards_path
+      redirect_to deck_cards_path, notice: 'Карта успешно обновленна'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.